src/components/PostDetails.tsx

```diff
@@ -1,107 +1,153 @@
-            Something went wrong
-          </div>
+export const PostDetails: React.FC<Props> = ({
+  openedPost,
+  postsComments,
+  onDeleteComments,
+  postBody,
+  postIdComment,
+  postTitle,
+  setPostsComments,
+}) => {
+  const [loading, setLoading] = useState(true);
+  const [writeCommentVisible, setWriteCommentVisible] = useState(false);
+  const [errorMessage, setErrorMessage] = useState<string | null>('');
 
-          <p className="title is-4" data-cy="NoCommentsMessage">
-            No comments yet
-          </p>
+  const loadPostComments = async (postId: number) => {
+    try {
+      setLoading(true);
 
-          <p className="title is-4">Comments:</p>
+      const comments = await getComments(postId);
 
-          <article className="message is-small" data-cy="Comment">
-            <div className="message-header">
-              <a href="mailto:misha@mate.academy" data-cy="CommentAuthor">
-                Misha Hrynko
-              </a>
-              <button
-                data-cy="CommentDelete"
-                type="button"
-                className="delete is-small"
-                aria-label="delete"
-              >
-                delete button
-              </button>
-            </div>
+      setPostsComments(comments);
+      setErrorMessage(null);
+    } catch (error) {
+      setPostsComments([]);
+      setErrorMessage('Failed to load comments.');
+    } finally {
+      setLoading(false);
+    }
+  };
 
-            <div className="message-body" data-cy="CommentBody">
-              Some comment
-            </div>
-          </article>
+  const toggleWriteComment = () => {
+    setWriteCommentVisible(prevState => !prevState);
+  };
 
-          <article className="message is-small" data-cy="Comment">
-            <div className="message-header">
-              <a href="mailto:misha@mate.academy" data-cy="CommentAuthor">
-                Misha Hrynko
-              </a>
+  const addNewComment = (comment: Comment) => {
+    setPostsComments(prevComments => [comment, ...prevComments]);
+    setWriteCommentVisible(false);
+  };
 
-              <button
-                data-cy="CommentDelete"
-                type="button"
-                className="delete is-small"
-                aria-label="delete"
-              >
-                delete button
-              </button>
-            </div>
-            <div className="message-body" data-cy="CommentBody">
-              One more comment
-            </div>
-          </article>
+  useEffect(() => {
+    setWriteCommentVisible(false);
+  }, [openedPost]);
 
-          <article className="message is-small" data-cy="Comment">
-            <div className="message-header">
-              <a href="mailto:misha@mate.academy" data-cy="CommentAuthor">
-                Misha Hrynko
-              </a>
+  useEffect(() => {
+    if (openedPost && openedPost.id) {
+      loadPostComments(openedPost.id);
+    }
+  }, [openedPost.id]);
```

Comment:
Fix it 
<img width="1183" alt="image" src="https://github.com/mate-academy/react_dynamic-list-of-posts/assets/64750958/36879832-a591-455a-9ae0-2c3ce1575bb5">